events/events.go

```diff
@@ -0,0 +1,379 @@
+package events
+
+import (
+	json "encoding/json"
+	"errors"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+	manifold "github.com/manifoldco/go-manifold"
+	merrors "github.com/manifoldco/go-manifold/errors"
+	"github.com/manifoldco/go-manifold/idtype"
+)
+
+// Type represents the different types of events.
+type Type string
+
+const (
+	// TypeUserCreated represents a user creation
+	TypeUserCreated Type = "user.created"
+
+	// TypeOperationProvisioned represents a provision operation
+	TypeOperationProvisioned Type = "operation.provisioned"
+
+	// TypeResourceCreated represents a resource creation
+	TypeResourceCreated Type = "resource.created"
+)
+
+// State represents the state of an event. Events usually starts only
+// containing raw references to the system and later are processed to expand
+// objects information.
+type State string
+
+const (
+	// StatePending represents the event information is pending expansion. That's
+	// done as a worker job. Users don't have access while an event is pending.
+	StatePending State = "pending"
+
+	// StateTracking represents the event is sending information to a 3rd-party
+	// analytics. Users can access an event while is tracking state.
+	StateTracking State = "tracking"
+
+	// StateDone represents the event information expansion is done. The event
+	// is immutable going forward. Users can access a done event.
```

Comment:
consider `complete event.` or `completed event.` instead

Reply:
All other jobs (provision, billing) end with done, that's why I used the same name.

Reply:
or are you talking about the comment? :D

Reply:
yeah just the comment :D it's fine though, just read weird to me.